payroll/views.py

```diff
@@ -90,46 +115,88 @@ def get_context_data(self, **kwargs):
             source_link = None
 
         context.update({
-            'jobs': Job.of_employer(self.object.id, n=5),
-            'median_salary': self.median_entity_salary(),
+            'jobs': self.jobs,
+            'median_salary': self.median_entity_salary,
             'headcount': len(employee_salaries),
             'total_expenditure': sum(employee_salaries),
-            'salary_percentile': self.salary_percentile(),
-            'expenditure_percentile': self.expenditure_percentile(),
+            'salary_percentile': self.salary_percentile,
+            'expenditure_percentile': self.expenditure_percentile,
             'employee_salary_json': json.dumps(binned_employee_salaries),
             'data_year': 2017,
             'source_link': source_link,
         })
 
         return context
 
-    def median_entity_salary(self):
+    def get_median_entity_salary(self):
```

Comment:
Renamed the original function to a more explicit getter, used by the cached property.

Would it be cleaner to refactor this into the cached property itself?

Reply:
i'm fine with it as written! it's nice to have the longer logic broken out.